Release Notes PMM 2.32.0 (#904)
percona/pmm-doc: docs/release-notes/2.32.0.md

- [PMM-9545](https://jira.percona.com/browse/PMM-9545): For effective monitoring we have added an additional metric for k8s cluster added in PMM.
 
## Improvements

- [PMM-10156](https://jira.percona.com/browse/PMM-10156): Starting with PMM 2.32.0, we have an enhanced PMM dashboard that gives a bird-view of the services, infrastructure, and critical issues (if any). Thus, you can effortlessly diagnose the problem from the home dashboard with just one click.

  
- [PMM-9946](https://jira.percona.com/browse/PMM-9946): To enhance user experience, we have simplified the deployment of PMM inside the K8s cluster using Manifest or Helm.

- [PMM-10393](https://jira.percona.com/browse/PMM-10393): Panel for total time taken for scraping metrics by Postgres exporter is improved. Instead of showing coarse-grained measurements of the total time taken by the exporter, it now shows the total time taken per collector.

> - [PMM-10555](https://jira.percona.com/browse/PMM-10555): Add meta-metric support

Author: @ShashankSinha252 @denisok Could you please provide description for this ticket.

Reviewer: Runtime for collectors in MongoDB exporter is pushed via Telemetry to PMM portal service. This information is used to detect unusually large scrape time for collector and improve collector performance in future releases.

# Percona Monitoring and Management 2.32.0

| **Release date:** | Nov 7, 2022                                                                                    |
| ----------------- | ----------------------------------------------------------------------------------------------- |
| **Installation:** | [Installing Percona Monitoring and Management](https://www.percona.com/software/pmm/quickstart) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

!!! caution alert alert-warning "Important"
    We recommend using the latest version of PMM. This ensures that you have access to the latest PMM features and that your environment runs on the latest version of the underlying components, such as VictoriaMetrics, with all the bug fixes in place.

## Release Highlights

### State-of-the-art new Home dashboard

In our constant effort to enhance user experience in PMM and considering your feedback, we have added another feather to our cap by replacing the existing home dashboard with a brand new Home dashboard in PMM 2.32.0.  

> This new experimental dashboard will be shipped to the PMM users. This dashboard will be uploaded to the Experimental folder to enable the users to test them in their environment and provide feedback.

Reviewer 2: @rasika-chivate we aren't uploading it to the Experimental dashboard. This paragraph is not correct :) 
We may say something like this: 
We're releasing this dashboard as your default home dashboard which was released as experimental dashboards in 2.30. Thanks to your feedback, it is now released with an improved version in 2.32.

Author: Done

- Improved visualization of the Used Memory Anomaly Panel
- Added a unit (%) to the CPU Anomalies panel for precision
- Added and refined tooltips for all the panels on the dashboard
- Enhanced the High CPU Usage, High Disk Queue, and High Memory Usage panels to make the troubleshooting process effortless and more actionable. Just click the node name on the panel to see the details of the problem.
- Enabled sparklines on the Disk Read and Write panels to represent and show trends in your data visually.
 

### Enhanced main menu

- In PMM 2.32.0, you can easily access all dashboards for monitored services due to the contemporized navigation structure. All the dashboards in the services folder that were not accessible from the main menu are now accessible from the main menu via **Other Dashboards**. Also, your custom dashboards can be accessed easily from the main menu via **Other Dashboards**

    For example, to access the dashboards for MYSQL, go to *MYSQL >> Other dashboards*.

- Only monitored services are now included in the main menu, enhancing the user experience and minimizing cognitive load to maximize usability.

> 

Reviewer 2: Can we add an example here? e.g. If you are only monitoring MySQL service, main menu will present MySQL dashboards instead of listing all dashboards that you're not monitoring.

Author: Done

## New Features

- [PMM-8792](https://jira.percona.com/browse/PMM-8792), [PMM-9055](https://jira.percona.com/browse/PMM-9055), [PMM-9056](https://jira.percona.com/browse/PMM-9056): PMM 2.32.0 introduces the parameter --max-query-length in the pmm-admin command for MYSQL, PostreSQL, and MongoDB databases. Now, queries longer than 2048 characters are not truncated (if parameter is set -1, which means no limit, or value is bigger than 2048), and the Explain and Examples tabs on the QAN dashboard do not throw an error [PMM-1899](https://jira.percona.com/browse/PMM-1899). These tabs provide contextual information pertaining to the query, thus enhancing user experience.

> 

Reviewer 2: PMM-1899 - is it in the same paragraph with PMM-8792, is it correct?

Author: yes.

> - [PMM-10579](https://jira.percona.com/browse/PMM-10579): We have expanded and improved the main menu for PMM 2.31.0 based on Grafana 9.1. The main menu contains crucial elements, one of them being starred dashboards.

Reviewer 3: 2.32.0 instead 2.31.0 here?

Author: Updated.

- Only monitored services are now included in the main menu, enhancing the user experience and minimizing cognitive load to maximize usability.

    For example, you will only see MongoDB dashboards in the main menu if you monitor the MongoDB database rather than all the other dashboards.

## New Features

- [PMM-9366](https://jira.percona.com/browse/PMM-9366): [Backup Management]: Added possibility to restore MongoDB PITR backups. This option is available under **Backup > All Backups > Actions > Restore this backup** for any PITR-type backups available in the table.
- [PMM-9368](https://jira.percona.com/browse/PMM-9368): [Backup Management]: Added support for local client storage: When adding storage locations under **Backup > Storage Location**, you can now also select **Local Client** option as the location type for storing future backups.
- [PMM-10599](https://jira.percona.com/browse/PMM-10599), [PMM-10503](https://jira.percona.com/browse/PMM-10599): [Backup Management]:  Refactored User Interface for creating backups to enable you to create on-demand and scheduled backups from the same window instead of using separate modals. 

> - [PMM-10470](https://jira.percona.com/browse/PMM-10470): [Backup Management]: Following the support for MongoDB physical backups introduced in PMM 2.31, we have now added the possibility to restore these backups.  Restoring MongoDBphysical backups requires additional manual operations after the restore and is only available with Percona Server for MongoDB.

Reviewer 3: add space here pls "MongoDBphysical"

- [PMM-10502](https://jira.percona.com/browse/PMM-10502), [PMM-10831](https://jira.percona.com/browse/PMM-10831): [Backup Management]: Improved User Experience and labels on all four tabs of the Backup section: 
    - Added icons to better illustrate failed and successful backups.
    - Added **Service Name** column to show the service from where the backup was retrieved.
    - Renamed **DB Vendor** tab to **DB Technology**.
    - Improved **Actions** menu with clearer labeled options.

- [PMM-10742](https://jira.percona.com/browse/PMM-10742): [Backup Management]: Extended PMM compatibility with Percona Backup for MongoDB (PMB) to support the latest version, v. 2.0 PBM.
- [PMM-10829](https://jira.percona.com/browse/PMM-10829): [Backup Management]: The backups table under **Backup > All Backups** now includes a new **Location** column which displays the service where the backup came from (S3 or Local Storage).

 

> - [PMM-8792](https://jira.percona.com/browse/PMM-8792), [PMM-9055](https://jira.percona.com/browse/PMM-9055), [PMM-9056](https://jira.percona.com/browse/PMM-9056): PMM 2.32.0 introduces the parameter `--max-query-length` in the `pmm-admin` command for MYSQL, PostgreSQL, and MongoDB databases. It can be also setup through UI. Now, queries longer than 2048 characters are not truncated (if the parameter is set to `-1`, which means no limit or value greater than 2048 chars), and the **Explain** and **Examples** tabs on the QAN dashboard do not throw an error, see [PMM-1899](https://jira.percona.com/browse/PMM-1899). These tabs are important as they provide contextual information pertaining to the query.

Reviewer 4: ```suggestion
- [PMM-8792](https://jira.percona.com/browse/PMM-8792),  [PMM-9055](https://jira.percona.com/browse/PMM-9055), [PMM-9056](https://jira.percona.com/browse/PMM-9056) - PMM 2.32.0 introduces the parameter --max-query-length in the pmm-admin command for MYSQL, PostreSQL, and MongoDB databases. It can be also setup through UI during adding of service. You can see new input in form, where you can setup you value of max query length. With these changes, queries longer than 2048 characters are not truncated (if parameter is set -1, which means no limit, or value is bigger than 2048), and the Explain and Examples tabs on the QAN dashboard do not throw an error ([PMM-1899](https://jira.percona.com/browse/PMM-1899)). These tabs provide contextual information pertaining to the query, thus enhancing user experience.
```

- [PMM-10502](https://jira.percona.com/browse/PMM-10502), [PMM-10831](https://jira.percona.com/browse/PMM-10831): [Backup Management]: Improved User Experience and labels on all four tabs of the Backup section: 
  - Added icons to better illustrate failed and successful backups.
  - Added **Service Name** column to show the service from where the backup was retrieved.
  - Renamed **DB Vendor** tab to **DB Technology**.
  - Improved **Actions** menu with clearer labeled options.

- [PMM-10742](https://jira.percona.com/browse/PMM-10742): [Backup Management]: Extended PMM compatibility with Percona Backup for MongoDB (PMB) to support the latest version, v. 2.0 PBM.
- [PMM-10829](https://jira.percona.com/browse/PMM-10829): [Backup Management]: The backups table under Backup > All Backups now includes a new Location column which displays the service where the backup came from (S3 or Local Storage).

 

> - [PMM-8792](https://jira.percona.com/browse/PMM-8792), [PMM-9055](https://jira.percona.com/browse/PMM-9055), [PMM-9056](https://jira.percona.com/browse/PMM-9056): PMM 2.32.0 introduces the parameter `--max-query-length` in the `pmm-admin` command for MYSQL, PostgreSQL, and MongoDB databases. Now, queries longer than 2048 characters are not truncated (if the parameter is set to `-1`, which means no limit or value greater than 2048 chars), and the **Explain** and **Examples** tabs on the QAN dashboard do not throw an error, see [PMM-1899](https://jira.percona.com/browse/PMM-1899). These tabs are important as they provide contextual information pertaining to the query.

Reviewer 4: Its also possible to do this in UI, could we also mention it somehow?

> - [PMM-8792](https://jira.percona.com/browse/PMM-8792), [PMM-9055](https://jira.percona.com/browse/PMM-9055), [PMM-9056](https://jira.percona.com/browse/PMM-9056): PMM 2.32.0 introduces the parameter `--max-query-length` in the `pmm-admin` command for MYSQL, PostgreSQL, and MongoDB databases. Now, queries longer than 2048 characters are not truncated (if the parameter is set to `-1`, which means no limit or value greater than 2048 chars), and the **Explain** and **Examples** tabs on the QAN dashboard do not throw an error, see [PMM-1899](https://jira.percona.com/browse/PMM-1899). These tabs are important as they provide contextual information pertaining to the query.

Reviewer 4: ```suggestion
- [PMM-8792](https://jira.percona.com/browse/PMM-8792), [PMM-9055](https://jira.percona.com/browse/PMM-9055), [PMM-9056](https://jira.percona.com/browse/PMM-9056): PMM 2.32.0 introduces the parameter `--max-query-length` in the `pmm-admin` command for MYSQL, PostgreSQL, and MongoDB databases. It can be also setup through UI. Now, queries longer than 2048 characters are not truncated (if the parameter is set to `-1`, which means no limit or value greater than 2048 chars), and the **Explain** and **Examples** tabs on the QAN dashboard do not throw an error, see [PMM-1899](https://jira.percona.com/browse/PMM-1899). These tabs are important as they provide contextual information pertaining to the query.
```

- [PMM-9545](https://jira.percona.com/browse/PMM-9545): To monitor Kubernetes clusters, PMM now collects Kubernetes metrics from kube-state-metrics. For more information on kube-state-metrics, see  [documentation](https://github.com/kubernetes/kube-state-metrics).
    
    !!! caution alert alert-warning "Disclaimer"
        This feature is still [technical preview](../details/glossary.md#technical-preview) and is subject to change. We recommend that early adopters use this feature for testing purposes only.

- [PMM-10579](https://jira.percona.com/browse/PMM-10579): Only monitored services are displayed on the main menu for a clean and simple User Interface.
- [PMM-10512](https://jira.percona.com/browse/PMM-10512): [Alerting]:  When creating new alert rules. you can now use two new labels (**application_name** and **usename**) for **pg_stat_activity_** metrics.

## Improvements

  - [PMM-10156](https://jira.percona.com/browse/PMM-10156): Starting with PMM 2.32.0, we have an enhanced PMM dashboard that gives a bird-view of the services, infrastructure, and critical issues (if any). Thus, you can effortlessly diagnose the problem from the home dashboard with just one click.

  

> - [PMM-9946](https://jira.percona.com/browse/PMM-9946): To enhance user experience, we have simplified the deployment of PMM inside the K8s cluster using Manifest or Helm.

Reviewer 5: To enhance user experience, once PMM is deployed to a Kubernetes cluster and a user wants to use DBaaS, PMM automatically registers that Kubernetes cluster to DBaaS.

    We recommend using the latest version of PMM. This ensures that you have access to the latest PMM features and that your environment runs on the latest version of the underlying components, such as VictoriaMetrics, with all the bug fixes in place.

## Release Highlights

### MongoDB Backup Management is now generally available

We are happy to announce that the Backup Management for MongoDB has reached General Availability and is no longer in Technical Preview.

This change comes with significant functionality enhancements and added UX improvements:

- **Point-in-time recovery (PITR) restores for MongoDB backups**: This enables you to restore Amazon S3 backups from any checkpoint in the past.
- **Local client storage**: In addition to Amazon S3, we have now also enabled the **Local Client** location option, which saves backups on the remote folder mounted to PMM client host.  
- **Key UX improvements**: You can now create on-demand and scheduled backups from the same window instead of using separate modals.
    In addition, we have added more icons and renamed tabs columns to better label the actions available behind important options.

> #### Supported setups
> 
> MongoDB Backup Management now supports replica set setups for the following actions:
> 
> - Create logical snapshot backups
> - Create logical PITR backups
> - Create physical snapshot backups. This is available only with Percona Server for MongoDB.
> - Restore logical snapshot backups
> - Restore physical backups. This requires additional manual operations after the restore and is only available with Percona Server for MongoDB.
> - Restore logical PITR backups from S3.
>   
> #### Current limitations
> 
> - Restoring logical PITR backups only supports S3 storage type.
> - Restoring physical backups requires manual post-restore actions.
> - Restoring a MongoDB backup on a new cluster is not yet supported.
> - Restoring physical backups for containerized MongoDB setups is not supported.
> - Local storage for MySQL is not supported.
> - Sharded cluster setups are not supported.
> - Backups that are stored locally cannot be removed automatically.
> - Retention is not supported for PITR artifacts.
> 
> #### What’s next?
> 
> As you can well guess, our immediate plan for Backup Management is to continue to iterate on the current limitations listed above and make the restore processes as seamless as possible for all database types.
> 

Reviewer 6: ```suggestion

```

Let's keep this for the release blog post

Author: @catalinaadam Please take care of this...

- Color-coded states on panels for a quick visual representation of the problem areas
- Comprehensive and easy-to-use dashboard to reduce cognitive load

>  

Author: The home dashboard page needs to be updated first. I will add the link to the blog post.
